scrapers/ESP/index.js

```diff
@@ -1,38 +1,128 @@
-        if (rules.isAcceptable(data, null, this._reject)) {
-          states.push(data);
-        }
-      }
+    const ISO = DateTimeFormatter.ofPattern('yyyy-MM-dd');
+    const isDate = s => s.includes('/');
+
+    const parseDate = s => {
+      // Source data currently uses ddmmyyyy format for dates,
+      // but is (maybe?) planning to switch to yyyymmdd.
+      // See https://github.com/datadista/datasets/issues/14
+      // When/if this happens, replace contents of this function with:
+      // return LocalDate.parse(s)
+
+      const [d, m, y] = s.split('/');
+      return LocalDate.parse(`${y}-${m}-${d}`);
+    };
+
+    const mostRecent = dates => dates.sort().pop();
+
+    const rawData = {};
+    for (const { name, url } of this.sources) {
+      rawData[name] = await fetch.csv(url, false);
     }
-    states.push(transform.sumData(states));
-    return states;
+
+    // `data_raw` looks like this:
+    // ```js
+    // {
+    //    cases: [
+    //      {
+    //        cod_ine: '01',
+    //        CCAA: 'Andalucía',
+    //        '27/02/2020': '1',
+    //        '28/02/2020': '6',
+    //        // ...
+    //      },
+    //      {
+    //        cod_ine: '02',
+    //        CCAA: 'Aragón',
+    //        '27/02/2020': '0',
+    //        '28/02/2020': '1',
+    //        // ...
+    //      },
+    //      //... other locations
+    //    ],
+    //    recovered: [
+    //      // same format
+    //    ],
+    //    deaths: [
+    //      // same format
+    //    ]
+    // }
+    // ```
+
+    const data = rawData.cases
+      .filter(casesRow => casesRow.CCAA !== 'Total')
+      .flatMap(casesRow => {
+        const location = casesRow.CCAA;
+        const sameLocation = d => d.CCAA === location;
+        const deathsRow = rawData.deaths.find(sameLocation);
+        const recoveredRow = rawData.recovered.find(sameLocation);
+        return Object.keys(casesRow)
+          .filter(isDate)
+          .map(date => {
+            return {
+              state: location,
+              date: parseDate(date).format(ISO),
+              cases: +casesRow[date] || 0,
+              deaths: +deathsRow[date] || 0,
+              recovered: +recoveredRow[date] || 0
+            };
+          });
+      });
+
+    // `data` is an array of objects that look like this:
+    // ```js
+    // {
+    //   location: 'Andalucía',
+    //   date: '2020-02-27',
+    //   cases: 1,
+    //   deaths: 0,
+    //   recovered: 0
+    // },
+    // ```
+
+    const scrapeDate = process.env.SCRAPE_DATE ? LocalDate.of(...process.env.SCRAPE_DATE.split('-')) : undefined;
+
+    const sampleRow = rawData.cases[0];
+    const dates = Object.keys(sampleRow)
+      .filter(isDate)
+      .map(parseDate);
+    const latestDate = mostRecent(dates);
+
+    // use the scrape date, or the latest available
+    const date = (scrapeDate || latestDate).format(ISO);
+
+    // return data from that date
+    const locations = data.filter(d => d.date === date);
```

Comment:
In and around here is where you need the logic to handle the errors for requested date out of range.